instructure/canvas-ios
# Improve assignment list

What's new?
- Assignment filtering was re-written to use on device filtering solving the issue of the last selected filter overwriting/deleting other assignments saved for offline mode.
- Since all assignments are available offline I enabled the grading period selector in this mode on the assignment list screen.
- I fixed a UI issue of section headers having an extra spacing above them by using ScrollView+LazyVStack instead of List.
- Added a reactive exhaust method to API.

refs: MBL-17208
affects: Student
release note: Updated assignments list to allow grading period filter to work in offline mode.

test plan:
- See ticket for issue description.
- Clean your device data.
- Sync a course with multiple grading periods.
- Without entering the assignment list in online mode turn on offline mode.
- Visit assignment details in offline mode.
- Grading period filter should work. Assignment list should match the web for each grading period.
- Go online. Switch grading periods.
- Restart the app in offline mode.
- Check if assignment list still displays the expected assignments.
- Test if grades list still works as expected in terms of displayed assignments and grading periods.
- Test assignment list and grades list without any grading periods.

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td><img src="https://github.com/instructure/canvas-ios/assets/72396990/d1061c64-e95e-4c0d-a9ab-3077f79d619d" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/72396990/e35bc06c-4b64-4820-b4e8-37ed36ea74f5" maxHeight=500></td>
</tr>
</table>

## Checklist
